### PR TITLE
Swagger perSite and perListener firewall policy - 2019_09_01

### PR DESCRIPTION
**Pull request by venk8**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json [hunk 1]
```diff
@@ -306,6 +306,22 @@
           },
           "description": "A collection of references to application gateways."
         },
+        "httpListeners": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "./applicationGateway.json#/definitions/ApplicationGatewayHttpListener"
+          },
+          "description": "A collection of references to application gateway httplisteners."
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json]:
```suggestion
          "description": "A collection of references to application gateway http listeners."
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json [hunk 2]
```diff
@@ -306,6 +306,22 @@
...
+        },
+        "pathBasedRules": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "./applicationGateway.json#/definitions/ApplicationGatewayPathRule"
+          },
+          "description": "A collection of references to application gateway pathrules."
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json]:
```suggestion
          "description": "A collection of references to application gateway path rules."
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json [hunk 3]
```diff
@@ -306,6 +306,22 @@
...
+          "items": {
+            "$ref": "./applicationGateway.json#/definitions/ApplicationGatewayHttpListener"
+          },
+          "description": "A collection of references to application gateway httplisteners."
+        },
+        "pathBasedRules": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "./applicationGateway.json#/definitions/ApplicationGatewayPathRule"
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json]: If this is a list of references it should be of type `./network.json#/definitions/SubResource`

File: specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json [hunk 4]
```diff
@@ -306,6 +306,22 @@
           },
           "description": "A collection of references to application gateways."
         },
+        "httpListeners": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "./applicationGateway.json#/definitions/ApplicationGatewayHttpListener"
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json]: If this is a list of references it should be of type `./network.json#/definitions/SubResource`

**Review by anton-evseev — CHANGES_REQUESTED** [review]
Please also rebase your branch to remove unrelated commits (52 is too much)

File: specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json (hunk 4 above)
Comment (venkatsvpr) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json]: addressed the comment.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json [hunk 5]
```diff
@@ -306,6 +306,22 @@
           },
           "description": "A collection of references to application gateways."
         },
+        "httpListeners": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "./applicationGateway.json#/definitions/SubResource"
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json]:
```suggestion
            "$ref": "./network.json#/definitions/SubResource"
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json [hunk 6]
```diff
@@ -306,6 +306,22 @@
...
+          },
+          "description": "A collection of references to application gateway http listeners."
+        },
+        "pathBasedRules": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "./applicationGateway.json#/definitions/SubResource"
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/webapplicationfirewall.json]:
```suggestion
            "$ref": "./network.json#/definitions/SubResource"
```

**Review by anton-evseev — CHANGES_REQUESTED** [review]

**Review by KrisBash — APPROVED** [review]
LGTM

**Review by anton-evseev — APPROVED** [review]

**Comment by number213** [comment]
@tjprescott could you please review and merge?

**Review by tjprescott — APPROVED** [review]

**Pull request merged**